src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +91,11 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
```

Comment:
I think we need to make `CopyFrom` itself no-op. I don't want to risk a `NullReferenceException` being thrown from `CreateResponseHeader` and the like.